ksqldb-engine/src/main/java/io/confluent/ksql/engine/RuntimeAssignor.java

```diff
@@ -79,11 +80,15 @@ public String getRuntimeAndMaybeAddRuntime(final QueryId queryId,
     }
     runtimesToSources.get(runtime).addAll(sources);
     idToRuntime.put(queryId, runtime);
+    log.info("Assigning query {} to runtime {}", queryId, runtime);
     return runtime;
   }
 
   public void dropQuery(final PersistentQueryMetadata queryMetadata) {
     if (queryMetadata instanceof BinPackedPersistentQueryMetadataImpl) {
+      log.info("Unassigning query {} form runtime {}",
+          queryMetadata.getQueryId(),
+          idToRuntime.getOrDefault(queryMetadata.getQueryId(), "no runtime"));
```

Comment:
Hmm... could it ever happen that `idToRuntime` is empty after we have done the registration? Maybe we can just do `log(Unassign..)` if it is not empty, and if it is empty meaning the query has not be registered yet we log a different like `log(Dropping un-registered query...)`.

Reply:
This should not be possible and would indicate a bug. I like the idea of `log(Dropping un-registered query...)` I'll try that instead